pkg/gencosmosdb/cosmosdb/cosmosdb.go

```diff
@@ -98,36 +122,27 @@ func (c *databaseClient) do(ctx context.Context, method, path, resourceType, res
 
 	resp, err := c.hc.Do(req)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer func() {
 		resp.Body.Read(nil)
 		resp.Body.Close()
 	}()
 
-	if headers != nil {
```

Comment:
there must be clients who are going to read headers who now need to read resp.Headers - these need to be fixed up